### DISCUSSION TITLE
can I use @ or ~ alias in route.ts file?

### DISCUSSION THREAD

**Discussion by flysky9981** [remix-run/remix#10222] [Q&A]
I set the alias of "./app" to "~" in vite.config.js file using the code below:

`export default defineConfig({
  resolve: {
    alias: {
      "~": resolve(__dirname, "./app"),
    }
  },`

and using the alias in route.ts file as below:
`import {allRoutes} from "~/routes-custom";`

but when i " npm run dev" in the terminal, the "remix" says:

`23:11:56 [remix] Route config in "routes.ts" is invalid.

Error: Failed to load url ~/routes-custom (resolved id: ~/routes-custom) in D:/home4dev/devjoe-dir/remix-antd-admin-master/app/routes.ts. Does the file exist?
`

when I change the `import {allRoutes} from "~/routes-custom";` to `import {allRoutes} from "./routes-custom";` it works. so what's wrong with this?

**Comment by sergiodxa** [comment]
If I'm not wrong, the routes.ts file runs at the same env of the vite.config where import aliases are not supported.

**Reply by flysky9981** [reply]
seems true, I traced the call stack, vite.config.js was called twice when the devServer is running. so, I can only use absolute or relative path in this file just llike it was in the configured routes option in "vite.config.js" using "defineRoutes" function.

But when or whether the routes.ts will be put into the app's env? I thought it was.

**Reply by flysky9981** [reply]
and it is weird that root.tsx is in the env where "@" and "~" alias is supported.

**Reply by sergiodxa** [reply]
The routes.ts file is used by Remix to build the app, but it's not put into the app env.

The root.tsx is part of the app because it's another route.
